### ISSUE TITLE
Rug commands failing that require RugArchiveReader

### ISSUE THREAD

**Issue by antjkennedy** [atomist/rug-cli#167]
Hi All,

We are having issued running any rug commands that require the `RugArchiveReader` - `rug install`, etc.

This is also present in the docker image.

```bash
$ docker run --rm -t atomist-docker.jfrog.io/rug-cli:0.29.0 rug describe archive atomist-rugs:rug-editors -X
```

Output
```
Processing dependencies
  Downloading atomist-rugs/rug-editors/maven-metadata.xml ← rugs (525 bytes) succeeded
  Downloading atomist-rugs/rug-editors/0.24.0/rug-editors-0.24.0.pom ← rugs (635 bytes) succeeded
  Downloading com/atomist/rug/maven-metadata.xml ← rugs (1 kb) succeeded
  Downloading com/atomist/rug/0.24.0/rug-0.24.0.pom ← rugs (25 kb) succeeded
  Downloading com/atomist/rug/0.25.0/rug-0.25.0.pom ← rugs (25 kb) succeeded
  Downloading com/atomist/rug/0.23.0/rug-0.23.0.pom ← rugs (25 kb) succeeded
  Downloading com/atomist/atomist-bom/0.10.0/atomist-bom-0.10.0.pom ← rugs (2 kb) succeeded
  Downloading org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.pom ← central (3 kb) succeeded
  Downloading org/slf4j/jcl-over-slf4j/1.7.25/jcl-over-slf4j-1.7.25.pom ← central (959 bytes) succeeded
  Downloading ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.pom ← central (12 kb) succeeded
  Downloading ch/qos/logback/logback-access/1.2.3/logback-access-1.2.3.pom ← central (5 kb) succeeded
  Downloading org/slf4j/slf4j-parent/1.7.25/slf4j-parent-1.7.25.pom ← central (13 kb) succeeded
  Downloading ch/qos/logback/logback-parent/1.2.3/logback-parent-1.2.3.pom ← central (17 kb) succeeded
  Downloading com/atomist/artifact-source/0.18.1/artifact-source-0.18.1.pom ← rugs (8 kb) succeeded
  Downloading ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.pom ← central (4 kb) succeeded
  Downloading org/slf4j/slf4j-api/1.7.21/slf4j-api-1.7.21.pom ← central (2 kb) succeeded
  Downloading org/slf4j/jcl-over-slf4j/1.7.21/jcl-over-slf4j-1.7.21.pom ← central (963 bytes) succeeded
  Downloading ch/qos/logback/logback-classic/1.1.8/logback-classic-1.1.8.pom ← central (12 kb) succeeded
  Downloading org/slf4j/slf4j-parent/1.7.21/slf4j-parent-1.7.21.pom ← central (13 kb) succeeded
  Downloading ch/qos/logback/logback-parent/1.1.8/logback-parent-1.1.8.pom ← central (17 kb) succeeded
  Downloading ch/qos/logback/logback-core/1.1.8/logback-core-1.1.8.pom ← central (4 kb) succeeded
  Downloading org/slf4j/slf4j-api/1.7.22/slf4j-api-1.7.22.pom ← central (2 kb) succeeded
  Downloading org/slf4j/slf4j-parent/1.7.22/slf4j-parent-1.7.22.pom ← central (13 kb) succeeded
  Downloading commons-io/commons-io/2.4/commons-io-2.4.pom ← central (9 kb) succeeded
  Downloading org/apache/commons/commons-parent/25/commons-parent-25.pom ← central (47 kb) succeeded
  Downloading org/apache/apache/9/apache-9.pom ← central (14 kb) succeeded
  Downloading com/fasterxml/jackson/core/jackson-annotations/2.8.7/jackson-annotations-2.8.7.pom ← central (1 kb) succeeded
  Downloading org/slf4j/jcl-over-slf4j/1.7.25/jcl-over-slf4j-1.7.25.jar ← central (16 kb) succeeded
  Downloading org/slf4j/slf4j-api/1.7.25/slf4j-api-1.7.25.jar ← central (40 kb) succeeded
  Downloading ch/qos/logback/logback-core/1.2.3/logback-core-1.2.3.jar ← central (460 kb) succeeded
  Downloading ch/qos/logback/logback-access/1.2.3/logback-access-1.2.3.jar ← central (104 kb) succeeded
  Downloading ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar ← central (283 kb) succeeded
  Downloading atomist-rugs/rug-editors/0.24.0/rug-editors-0.24.0-metadata.json ← rugs (18 kb) succeeded
  Downloading com/atomist/artifact-source/0.18.1/artifact-source-0.18.1.jar ← rugs (355 kb) succeeded
  Downloading atomist-rugs/rug-editors/0.24.0/rug-editors-0.24.0.zip ← rugs (669 kb) succeeded
  Downloading com/atomist/rug/0.25.0/rug-0.25.0.jar ← rugs (2 mb) succeeded
Resolving dependencies for atomist-rugs:rug-editors (latest·zip) completed
Loading atomist-rugs:rug-editors (0.24.0·zip) failed

java.lang.NoClassDefFoundError: com/atomist/project/archive/RugArchiveReader
        at com.atomist.rug.resolver.loader.ProvenanceAddingRugLoader.loadArtifact(ProvenanceAddingRugLoader.java:242)
        at com.atomist.rug.resolver.loader.ProvenanceAddingRugLoader.load(ProvenanceAddingRugLoader.java:137)
        at com.atomist.rug.cli.command.AbstractCompilingAndOperationLoadingCommand.doLoadRugs(AbstractCompilingAndOperationLoadingCommand.java:155)
        at com.atomist.rug.cli.command.AbstractCompilingAndOperationLoadingCommand.lambda$loadRugs$5(AbstractCompilingAndOperationLoadingCommand.java:184)
        at com.atomist.rug.cli.output.ProgressReportingOperationRunner.run(ProgressReportingOperationRunner.java:20)
        at com.atomist.rug.cli.command.AbstractCompilingAndOperationLoadingCommand.loadRugs(AbstractCompilingAndOperationLoadingCommand.java:183)
        at com.atomist.rug.cli.command.AbstractCompilingAndOperationLoadingCommand.run(AbstractCompilingAndOperationLoadingCommand.java:66)
        at com.atomist.rug.cli.command.AbstractCommand.run(AbstractCommand.java:47)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.atomist.rug.cli.command.ReflectiveCommandRunMethodRunner.invokeCommand(ReflectiveCommandRunMethodRunner.java:20)
        at com.atomist.rug.cli.command.ReflectiveCommandRunner.invokeReflectiveCommand(ReflectiveCommandRunner.java:100)
        at com.atomist.rug.cli.command.ReflectiveCommandRunner.invokeCommand(ReflectiveCommandRunner.java:250)
        at com.atomist.rug.cli.command.ReflectiveCommandRunner.runCommand(ReflectiveCommandRunner.java:61)
        at com.atomist.rug.cli.Runner.runCommand(Runner.java:93)
        at com.atomist.rug.cli.Runner.run(Runner.java:40)
        at com.atomist.rug.cli.Main.invokeRunner(Main.java:34)
        at com.atomist.rug.cli.Main.main(Main.java:25)
Caused by: java.lang.ClassNotFoundException: com.atomist.project.archive.RugArchiveReader
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        at com.atomist.rug.cli.classloading.DelegatingUrlClassLoader.loadClass(DelegatingUrlClassLoader.java:33)
        ... 20 more
```

**Comment by kipz** [comment]
Thanks @antjkennedy for making us aware of this problem. It looks like there is an incompatibility between the latest rug release (0.25.0) and the CLI.

Until we release a new CLI (hopefully later today), you can force the version of rug to 0.24.0 by adding `--requires 0.24.0` to your CLI commands.

You can also update the manifest.yml to force the version, but of course this only works if you're creating a new rug, and has the side-effect of pinning to an old version, which in the general case, won't be desirable!

**Comment by antjkennedy** [comment]
Excellent, thanks for the quick response

**Comment by cdupuis** [comment]
We have released rug-cli 0.30.2 yesterday. That version fixes this issue here.
